static/app/views/performance/database/noDataMessage.tsx

```diff
@@ -1,106 +1,117 @@
 import {Fragment} from 'react';
-import sumBy from 'lodash/sumBy';
 
 import ExternalLink from 'sentry/components/links/externalLink';
-import {t, tct} from 'sentry/locale';
+import {tct} from 'sentry/locale';
+import {Project} from 'sentry/types';
 import useOrganization from 'sentry/utils/useOrganization';
 import usePageFilters from 'sentry/utils/usePageFilters';
 import {normalizeUrl} from 'sentry/utils/withDomainRequired';
-import {useIneligibleProjects} from 'sentry/views/performance/database/useIneligibleProjects';
-import {useProjectSpanMetricCounts} from 'sentry/views/starfish/queries/useProjectSpanMetricsCounts';
+import {useDenylistedProjects} from 'sentry/views/performance/database/useDenylistedProjects';
+import {useOutdatedSDKProjects} from 'sentry/views/performance/database/useOutdatedSDKProjects';
 
 interface Props {
   Wrapper?: React.ComponentType;
+  isDataAvailable?: boolean;
 }
 
 function DivWrapper(props) {
   return <div {...props} />;
 }
 
-export function NoDataMessage({Wrapper = DivWrapper}: Props) {
+export function NoDataMessage({Wrapper = DivWrapper, isDataAvailable}: Props) {
   const {selection, isReady: pageFilterIsReady} = usePageFilters();
 
   const selectedProjectIds = selection.projects.map(projectId => projectId.toString());
 
-  const {data: projectSpanMetricsCounts, isLoading: areSpanMetricCountsLoading} =
-    useProjectSpanMetricCounts({
-      query: 'span.module:db',
-      statsPeriod: SAMPLE_STATS_PERIOD,
-      enabled: pageFilterIsReady,
+  const {projects: outdatedProjects, isFetching: areOutdatedProjectsFetching} =
+    useOutdatedSDKProjects({
       projectId: selectedProjectIds,
+      enabled: pageFilterIsReady && !isDataAvailable,
     });
 
-  const doesAnySelectedProjectHaveMetrics =
-    sumBy(projectSpanMetricsCounts, 'count()') > 0;
-
-  const {ineligibleProjects, isFetching: areIneligibleProjectsFetching} =
-    useIneligibleProjects({
+  const {projects: denylistedProjects, isFetching: areDenylistProjectsFetching} =
+    useDenylistedProjects({
       projectId: selectedProjectIds,
-      enabled: pageFilterIsReady && !doesAnySelectedProjectHaveMetrics,
     });
 
-  const organization = useOrganization();
-
-  const hasMoreIneligibleProjectsThanVisible =
-    ineligibleProjects.length > MAX_LISTED_PROJECTS;
-
-  if (areSpanMetricCountsLoading || areIneligibleProjectsFetching) {
+  // Hide message while fetching outdated projects or denylisted projects
+  if (areOutdatedProjectsFetching || areDenylistProjectsFetching) {
     return null;
   }
 
-  if (!projectSpanMetricsCounts) {
+  // Hide message if all conditions are satisfied
+  if (
+    isDataAvailable &&
+    !areOutdatedProjectsFetching &&
+    !areDenylistProjectsFetching &&
+    outdatedProjects.length === 0 &&
+    denylistedProjects.length === 0
```

Comment:
l: I'd prefer if we extract out these two conditions (`Hide message while fetching outdated projects or denylisted projects` and `Hide message if all conditions are satisfied) are extract into variables, and we have a single `return null`. This hopefully means that we can get rid of the comments as well.

Reply:
I totally get the extract into variables part, but why the single `return null`? Does that help readability in any way? Plus, if combined I have to evaluate _both_ conditions before `returning null`. Peanuts performance-wise, but still!

Reply:
The single return is more of a consequence of extract outing the variable, but I wouldn't be opposed to keeping it two different return conditions. I just would like the clarity the naming brings.

Reply:
Makes sense. I split the conditions more and grouped them up to improve meaning, should be better now!